pytest/common_lib/shared/metrics.py

```diff
@@ -14,8 +15,36 @@ class IntMetricName(str, Enum):
     MPC_PENDING_CKDS_QUEUE_ATTEMPTS_GENERATED = (
         "mpc_pending_ckds_queue_attempts_generated"
     )
+    MPC_CLUSTER_FAILED_SIGNATURES_COUNT = "mpc_cluster_failed_signatures_count"
 
 
 class DictMetricName(str, Enum):
     MPC_PEERS_INDEXER_BLOCK_HEIGHTS = "mpc_peers_indexer_block_heights"
     MPC_NETWORK_LIVE_CONNECTIONS = "mpc_network_live_connections"
+
+
+@dataclass
+class NodeMetrics:
+    queue_size: int
+    requests_indexed: int
+    responses_indexed: int
+    matching_responses_indexed: int
+    mpc_cluster_failed_signatures_count: int
+
+    def __sub__(self, other):
+        if isinstance(other, NodeMetrics):
+            res = NodeMetrics(0, 0, 0, 0, 0)
+            res.queue_size = self.queue_size - other.queue_size
+            res.requests_indexed = self.requests_indexed - other.requests_indexed
+            res.responses_indexed = self.responses_indexed - other.responses_indexed
+            res.matching_responses_indexed = (
+                self.matching_responses_indexed - other.matching_responses_indexed
+            )
+            res.mpc_cluster_failed_signatures_count = (
+                self.mpc_cluster_failed_signatures_count
+                - other.mpc_cluster_failed_signatures_count
+            )
+            return res
```

Comment:
Does this just return `None` if it is not an instance & do we risk silent errors?
What about the following
```suggestion
    def __sub__(self, other):
        if not isinstance(other, NodeMetrics):
            return NotImplemented
            
        res = NodeMetrics(0, 0, 0, 0, 0)
        res.queue_size = self.queue_size - other.queue_size
        res.requests_indexed = self.requests_indexed - other.requests_indexed
        res.responses_indexed = self.responses_indexed - other.responses_indexed
        res.matching_responses_indexed = (
            self.matching_responses_indexed - other.matching_responses_indexed
        )
        res.mpc_cluster_failed_signatures_count = (
            self.mpc_cluster_failed_signatures_count
            - other.mpc_cluster_failed_signatures_count
        )
        return res
```